fe_utils/finite_elements.py

```diff
@@ -18,8 +21,11 @@ def lagrange_points(cell, degree):
     <ex-lagrange-points>`.
 
     """
-
-    raise NotImplementedError
+    equi_lag_points = []
+    for i in product(range(degree + 1), repeat=cell.dim):  # if cell.dim=2, is all (i, j) combinations
+        if sum(i) <= degree:  # if cell.dim=1, trivially true
+            equi_lag_points.append(np.array(i) / degree)
```

Comment:
This is very nice. I haven't seen a truly dimension independent implementation of this before. As usual, it would be slightly neater and more Pythonic written as a list comprehension.